### PR TITLE
Add trim option to image processing and implement trimEdges function

### PR DESCRIPTION
**Pull request by vetler**
Adds option "trim" to automatically trim edges of the same color.

Before, w/o trim option:

test_image

After, with /trim/ option in URL:
output

### CODE REVIEW

File: transform.go [hunk 1]
```diff
@@ -309,5 +309,47 @@ func transformImage(m image.Image, opt Options) image.Image {
...
+	if opt.Trim {
+		m = trimEdges(m)
+	}
+
 	return m
 }
+
+func trimEdges(img image.Image) image.Image {
+	bounds := img.Bounds()
+	minX, minY, maxX, maxY := bounds.Max.X, bounds.Max.Y, bounds.Min.X, bounds.Min.Y
```
Comment (willnorris) [inline transform.go:322]: are you intentionally flipping min and max values here?

File: transform.go [hunk 2]
```diff
@@ -309,5 +309,47 @@ func transformImage(m image.Image, opt Options) image.Image {
...
+
+	// Get the color of the first pixel (top-left corner)
+	baseColor := img.At(bounds.Min.X, bounds.Min.Y)
+
+	// Check each pixel and find the bounding box of non-matching pixels
```
Comment (willnorris) [inline transform.go:327]: If we assume that the border will only be a small-ish percentage of the edge of the image, then we should be able to optimize this a bit such that we don't have to inspect every pixel.  Fine to leave as-is for now, particularly if we really focus on making sure there is good test coverage... then future optimizations are a lot safer to make.

**Review by willnorris — CHANGES_REQUESTED** [review]
This is really nicely done, thanks!  I've added some comments below, some just optimizations for future.  I'll take a look at trying to address this also, but wanted to go ahead and jot my notes down in case you get to it first.

**Comment by willnorris** [comment]
oh no... looks like my really long comment about tests didn't get saved!  I'll retype that.

File: transform_test.go [hunk 3]
```diff
@@ -375,3 +375,22 @@ func TestTransformImage(t *testing.T) {
 		}
 	}
 }
+
+func TestTrimBordersOfSameColor(t *testing.T) {
+	src := newImage(4, 4,
```
Comment (willnorris) [inline transform_test.go:380]: I think this warrants some table-driven tests, particular to cover edge cases.  What does this do for an empty image?  What about an image of a solid color?  (Does it trim nothing or everything?)

I also think a uniform image like this test case makes it easy to mask issues with swapped x and y values or min and max values.  We should have at least one test case of an irregular image.  I tried testing on locally and it failed.  It very well may have been an issue with my test, or it might have been the swapped min/max values noted above.

For example, I tried:

```
x x x x
x x o x
x o o x
x o o x
x x x x
x x x x
```

Thicker border on bottom than on top, and the second row with mixed values.  It should return:

```
x o
o o
o o
```

I'm not 100% sure what I actually got back, since the testing output it not terribly legible :)  I think I'll pull in a testing package that prints more legible diffs.

File: transform.go (hunk 1 above)
Comment (vetler) [reply transform.go:322]: Yeah, they are only initialized with min and max values, but after the loop minX/minY will be reduced from max to the smallest coordinates of non-matching pixels, and the opposite for maxY/minY (they will be increase from minimum to largest coordinates for non-matching pixels.

File: transform_test.go (hunk 3 above)
Comment (vetler) [reply transform_test.go:380]: I must admit I only tested the basic case with the red square, it didn't occurr to me to test with something more complicated 😅 Will look into it.

Comment (vetler) [reply transform_test.go:382]: @willnorris I added a test for this: https://github.com/willnorris/imageproxy/pull/441/commits/5a66c793b95bd3521993afe8d198eea2c0a0ae67

Also a few other variants, please take a look and let me know what you think!

**Comment by vetler** [comment]
I moved the trimming up to be the first transformation, as this makes more sense. If images are trimmed after scaling for instance, then the image will not be at expected size if it is trimmed.

**Comment by willnorris** [comment]
> @willnorris Are you waiting on anything more from me regarding this? Let me know if you think there's anything missing or any more tests that should be added

Not waiting on anything from you... just hadn't made the time to get back to look at this.  Will try to do so later today.

**Comment by willnorris** [comment]
I combined the tests to into a single table-driven test function, and did some other very minimal cleanup.  Merged as c361000.  Thanks for this!
